File: 4.4 Module VGF/vigo-feast-frontend/src/types/customer.ts
export interface Customer {
  customer_id: number;
  user_id: number | null;
  name: string;
  mobile: string | null;
  email: string | null;
  default_address_id: number | null;
  loyalty_points: number;
  franchisee_id: string | null;
  created_at: string;
  updated_at: string;
  is_active: boolean | null;
  is_deleted: boolean | null;
}

export interface CustomerListResponse {
  items: Customer[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: 4.4 Module VGF/vigo-feast-backend/app/schemas/customer.py
from datetime import date, datetime, time
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr


class CustomerBase(BaseModel):
    name: str
    mobile: Optional[str] = None
    email: Optional[EmailStr] = None
    default_address_id: Optional[int] = None
    loyalty_points: int = 0


class CustomerCreate(CustomerBase):
    user_id: Optional[int] = None
    franchisee_id: Optional[str] = None


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    mobile: Optional[str] = None
    email: Optional[EmailStr] = None
    default_address_id: Optional[int] = None
    loyalty_points: Optional[int] = None


class CustomerStatusUpdate(BaseModel):
    is_active: bool


class CustomerResponse(CustomerBase):
    model_config = ConfigDict(from_attributes=True)

    customer_id: int
    user_id: Optional[int] = None
    franchisee_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    is_active: Optional[bool] = True
    is_deleted: Optional[bool] = False


class CustomerListResponse(BaseModel):
    items: list[CustomerResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
